conn.go

```diff
@@ -99,6 +99,25 @@ const (
 	ApplicationVerification       VerifyResult = C.X509_V_ERR_APPLICATION_VERIFICATION
 )
 
+type VerifyError struct {
```

Comment:
Use the code below:

```
type verifyError struct {
    code VerifyResult
}

func (e verifyError) Error() string {
    return fmt.Sprintf("openssl verification error: %s", C.GoString(C.X509_verify_cert_error_string(C.long(e.code))))
}

func NewVerifyError(code VerifyResult) error {
 	if code == Ok {
 		return nil
 	}

 	return &VerifyError{code: code}
 }
```

Reply:
Do you mean: 
type VerifyError struct {
    code VerifyResult
}
... 
?

But what's about Code() method? I would like to leave Code() for some cases

Also what is a problem to call GetVerifyError() in Error()?